funding: enable push_msat

niftynei opened the pull request:
it's that time of year (merry xmas!) :santa: :mrs_claus: 

enables the ability to push_msat on fundchannel (and fundchannel_start)

rustyrussell commented:
Ack https://github.com/ElementsProject/lightning/commit/8919d9830de96d5eb14695101f23da3fa282cc12

niftynei commented:
re-applying ACK, added a Changelog line and fixed the documentation 

ACK https://github.com/ElementsProject/lightning/pull/3369/commits/6eaee6c263eb459a914a3de62b0749023d9b3aa2
